File: python_back/ecg_analyse_doctors.py
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy.exc import SQLAlchemyError
from sqlalchemy.future import select
from models import ECGAnalyseDoctors
import datetime


async def create_ecg_analyse_doctor(
    session: AsyncSession,
    ecg_analyse_id: int,
    doctor_id: int
) -> ECGAnalyseDoctors:
    new_ecg = ECGAnalyseDoctors(
        doctor_id=doctor_id,
        ecg_analyse_id=ecg_analyse_id,
        created_at=datetime.datetime.utcnow(),
        updated_at=datetime.datetime.utcnow()
    )
    session.add(new_ecg)
    session.commit()
    session.refresh(new_ecg)  # Get the ID after commit
    return new_ecg

File: backend/EkgAnalyzerApi/Models/ECGAnalyseDoctors.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EkgAnalyzerApi.Models
{
    [Table("ecg_analyse_doctors")]
    public class ECGAnalyseDoctors
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Column("ecg_analyse_id")]
        public int ECGAnalysesId { get; set; }   // MUHIM

        [ForeignKey(nameof(ECGAnalysesId))]
        public ECGAnalyses ECGAnalyse { get; set; } = null!;

        [Column("doctor_id")]
        public int DoctorId { get; set; }

        [Column("created_at")]
        public DateTime? CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updated_at")]
        public DateTime? UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}